rfc/rfc-0000-k8s-events.md

```diff
@@ -0,0 +1,176 @@
+machinations of the Cartographer controller, especially in debugging failures. For these situations, Kubernetes
+provides [events](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events)
+to expose the user to these temporal triggers/actions from inside controller processes.
+
+**Stamping** (creation of the resource definition from template and inputs) and **Submitting** (applying the resource
+definition to the cluster) are two very common events, with associated causes, that currently need further exposure to
+users.
+
+# What it is
+
+[what-it-is]: #what-it-is
+
+This first diagram shows the objects tracked or managed by the Cartographer controller, except for **Artifact** and
+**Controller Stamp**.
+
+![Cartographer Objects](./rfc-k8s-events/objects.jpg)
+
+## Developer View
+
+To date, Cartographer presents at least some information from all these sources to developers:
+
+![Owner](./rfc-k8s-events/owner.jpg)
+
+Note: `status.resources[.conditions]` do not currently look like this, but there is work to present a **Healthy** status
+[in this rfc](https://github.com/vmware-tanzu/cartographer/pull/738). Although there is no art to tackle sharing
+per-resource 'Submitted' status (I will raise a separate RFC), it's something that can be handled entirely using the
+status in a workload (ie, it does not require the use of events)
+
+With different initiatives (in RFCs for workload status improvements) the developer's view of information is becoming
+quite complete, but a key issue is the absence of awareness that "something is happening", especially after they've
+committed new code. With event's we can provide information that show's activity and causality not present in the
+current snapshot of state.
+
+We add events to the Owners (Workload/Deliverable) to enable debug and reasoning about Cartographer's behavior.
+
+## Operator View
+
+The situation is worse for users managing the supply chains (Devops/Ops). By looking at a supply chain, there is very
+little information about how supply chains are behaving, or if any issues are occurring for developers:
+
+![Blueprint](./rfc-k8s-events/blueprint.jpg)
+
+Although these users can access the logs, we can provide them a more concise view of how supply chains are behaving
+on-cluster.
+
+We add events to the Blueprints (Supply Chain/Delivery) to enable debug and reasoning about Cartographer's behavior.
+
+# How it Works
+
+[how-it-works]: #how-it-works
+
+To avoid over-communicating, and making the view of event's overwhelming and dilute for user's we will start with two
+key guiding principles:
+
+1. > Let it be something that isn't satisfactorily presented in other user views (eg: `k get workload`).
+2. > Let the event list be as meaningful as is necessary, and no more.
+
+The cartographer controller could emit the following events:
```

Comment:
Is this RFC proposing that these exact events exist, or that there will be events and here are some types of events that we think are interesting? It would be good to clarify that choice in the summary.

I'd tend to lean towards making these exemplar events and leaving the details a bit more up to the implementation. Minimally, allowing room for the message to evolve can be helpful.

Reply:
Did you miss this: https://github.com/vmware-tanzu/cartographer/blob/907cd2d949486463555c6908799a4048b9310e8a/rfc/rfc-0000-k8s-events.md?plain=1#L91-L92

Or is that not what you mean? Or is it perhaps not obvious enough?